test/odbc/connection_object.cpp

```diff
@@ -0,0 +1,44 @@
+#include "connection_object.h"
+#include "odbc_handler.h"
+
+ConnectionObject::ConnectionObject(string driver, string server, string port, string uid, string pwd, string dbname) {
+    db_driver_ = driver;
+    db_server_ = server;
+    db_port_ = port;
+    db_uid_ = uid;
+    db_pwd_ = pwd;
+    db_dbname_ = dbname;
+    
+    connection_string_ = "DRIVER={" + db_driver_ + "};SERVER=" + db_server_ + ";PORT=" + db_port_ + ";UID=" + db_uid_ + ";PWD=" + db_pwd_ + ";DATABASE=" + db_dbname_;
```

Comment:
What happens when you put an empty string for `Port` and `PWD`, is the string still valid or do we need an addition check to see if they're empty or not

Reply:
This logic should be in `Drivers::IsValidConnectionObject` in `drivers.cpp`. There, I check if there are no empty strings for driver, server, port, UID, & database. 

I omitted checking empty string for password because there may be no password for the database. Let me know if this sounds correct, or if there are other properties that can be empty but still form a valid connection string.

Reply:
👍 missed the part about it checking the port too, glanced over it too quickly.
Having nothing for password sounds correct to me, I'll give it a check later tad bit rusty on what can be empty or not